lib/media/track/localmediatrack.js

```diff
@@ -187,10 +194,10 @@ function restartWhenInadvertentlyStopped(localMediaTrack) {
   }
 
   document.addEventListener('visibilitychange', handleTrackStateChange);
-  localMediaTrack.on('stopped', handleTrackStateChange);
+  mediaStreamTrack.addEventListener('ended', handleTrackStateChange);
   return () => {
     document.removeEventListener('visibilitychange', handleTrackStateChange);
-    localMediaTrack.removeListener('stopped', handleTrackStateChange);
+    mediaStreamTrack.removeEventListener('ended', onended);
```

Comment:
```suggestion
    mediaStreamTrack.removeEventListener('ended', handleTrackStateChange);
```

Reply:
Thanks for catching this!

Reply:
but since this function accesses `mediaStreamTrack` it needs to be most recent. and therefore its better that its `let` instead of `const`